analyses/molecular-subtyping-EPN/02_ependymoma_generate_all_data.py

```diff
@@ -176,9 +177,20 @@ def fill_df_with_fpkm_zscores(df, fpkmdf, gene_name):
 
 # Adding Z-scores to dataframe
 expression_cols = ["RELA", "L1CAM", "ARL4D", "CLDN1", "CXorf67", "TKTL1", "GPBP1", "IFT46"]
-for gene in expression_cols:
-    EPN_notebook = fill_df_with_fpkm_zscores(EPN_notebook, fpkm_df, gene)
+# samples with rna matched with dna
+EPN_notebook_rna_dna = EPN_notebook[EPN_notebook["Kids_First_Biospecimen_ID_RNA"].notna()]
+# samples with only dna
+EPN_notebook_dna_only = EPN_notebook[EPN_notebook["Kids_First_Biospecimen_ID_RNA"].isna()]
 
+for gene in expression_cols:
+    # rna matched dna samples
+    EPN_notebook_rna_dna = fill_df_with_fpkm_zscores(EPN_notebook_rna_dna, fpkm_df, gene)
+    column_name = gene + "_expr_zscore"
+    # add NA to expression zscore columns for dna only samples
+    EPN_notebook_dna_only.loc[EPN_notebook_dna_only["Kids_First_Biospecimen_ID_RNA"].isna(),column_name] = np.nan
+    
+# append dataframes    
+EPN_notebook = EPN_notebook_rna_dna.append(EPN_notebook_dna_only)
 
 # Replacing all Nan values with NA so they are not empty when writing to a file
 EPN_notebook = EPN_notebook.replace(np.nan, 'NA', regex=True)
```

Comment:
Just noting here that in the two output files included here, `EPN_all_data.tsv` uses`NA`, but `EPN_all_data_withsubgroup.tsv` uses blank fields. I like the explicit `NA` here, but this should maybe be added to the other output?

Reply:
Thanks for the review @jashapiro ! I added some commits below to work on some of your suggestions and also added NA to EPN_all_data_withsubgroup.tsv.